ci/test_narwhals.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+# Copyright (c) 2025, NVIDIA CORPORATION.
+
+# Support invoking test_python_cudf.sh outside the script directory
+cd "$(dirname "$(realpath "${BASH_SOURCE[0]}")")"/../ || exit
+
+# Common setup steps shared by Python test jobs
+source ./ci/test_python_common.sh test_python_narwhals
+
+RAPIDS_VERSION="$(rapids-version)"
+
+rapids-mamba-retry install \
+  --channel "${CPP_CHANNEL}" \
+  --channel "${PYTHON_CHANNEL}" \
+  "cudf-polars=${RAPIDS_VERSION}"
+
+rapids-logger "Check GPU usage"
+nvidia-smi
+rapids-print-env
+EXITCODE=0
+trap "EXITCODE=1" ERR
+set +e
+
+rapids-logger "pytest narwhals"
+git clone https://github.com/narwhals-dev/narwhals --depth=1
+pushd narwhals || exit
+pip install -U -e ".[dev]"
```

Comment:
```suggestion
python -m pip install -U -e ".[dev]"
```

just for a bit more confidence we aren't crossing envs

Reply:
Good idea. I'm going to use `rapids-pip-retry` instead, which already uses `python -m pip`.